### PR TITLE
feat(loading)!: add MongoDB exporter

### PR DESCRIPTION
**Pull request by andrea-acampora**

### CODE REVIEW

File: alchemist-loading/src/main/kotlin/it/unibo/alchemist/loader/export/Extractor.kt [hunk 1]
```diff
@@ -36,12 +36,11 @@
...
 
     /**
-     * @return the name of the properties that this {@link Extractor} can
-     *         provide
+     * @return the name of the properties that this [Extractor] can
+     * provide
      */
-    List<String> getNames();
-
+    fun getColumnNames(): List<String>
```
Comment (DanySK) [inline alchemist-loading/src/main/kotlin/it/unibo/alchemist/loader/export/Extractor.kt:45]: val

File: alchemist-loading/src/main/kotlin/it/unibo/alchemist/loader/export/exporters/MongoDBExporter.kt [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        mongoService.stopService()
+    }
+
+    private fun convertToDocument(env: Environment<T, P>, reaction: Reaction<T>?, time: Time, step: Long): Document {
+        val document = Document()
+        dataExtractors.stream()
+            .forEach {
+                it.extractData(env, reaction, time, step).entries.forEach {
+                    entry ->
+                    document.append(entry.key, entry.value)
```
Comment (DanySK) [inline alchemist-loading/src/main/kotlin/it/unibo/alchemist/loader/export/exporters/MongoDBExporter.kt:69]: use a destructuring declaration

**Review by DanySK — CHANGES_REQUESTED** [review]
8/30

File: alchemist-loading/src/main/kotlin/it/unibo/alchemist/loader/export/exporters/MongoDBExporter.kt (hunk 2 above)
Comment (andrea-acampora) [reply alchemist-loading/src/main/kotlin/it/unibo/alchemist/loader/export/exporters/MongoDBExporter.kt]: Done

File: alchemist-loading/src/main/kotlin/it/unibo/alchemist/loader/export/extractors/ExecutionTime.kt [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import it.unibo.alchemist.loader.export.Extractor
+import it.unibo.alchemist.model.interfaces.Environment
+import it.unibo.alchemist.model.interfaces.Reaction
+import it.unibo.alchemist.model.interfaces.Time
+
+/**
+ * An extractor which provides informations about the running time of the simulation.
+ *
+ */
+class ExecutionTime : Extractor<Double> {
```
Comment (DanySK) [inline alchemist-loading/src/main/kotlin/it/unibo/alchemist/loader/export/extractors/ExecutionTime.kt:21]: Wait, what is the scope of this?
I hope it's not meant to be logged when `time` is required, is it?

File: alchemist-loading/src/test/resources/testExporters.yml [hunk 4]
```diff
@@ -7,3 +7,11 @@ export:
     data:
       - time
       - molecule: "default_module:default_program"
+
+  - type: MongoDBExporter
+    parameters:
+      uri: "mongodb+srv://alchemist-user:alchemist-pwd@alchemist.ewxds.mongodb.net/myFirstDatabase?retryWrites=true&w=majority"
```
Comment (DanySK) [inline alchemist-loading/src/test/resources/testExporters.yml:13]: this is fragile. Extremely fragile. And this is a password in clear, or am I wrong?

Comment (andrea-acampora) [reply alchemist-loading/src/test/resources/testExporters.yml]: Done: changed to a local instance of mongo.

File: alchemist-loading/src/main/kotlin/it/unibo/alchemist/loader/export/extractors/ExecutionTime.kt (hunk 3 above)
Comment (andrea-acampora) [reply alchemist-loading/src/main/kotlin/it/unibo/alchemist/loader/export/extractors/ExecutionTime.kt:21]: No, it's a custom export and it logs the running time in seconds.
In order to log it you have to specify in the export section the type as `ExecutionTime`.

**Review by DanySK — APPROVED** [review]

**Pull request merged**